File: user-service/src/main/java/socialapp/userservice/model/dto/RegistrationDto.java
package socialapp.userservice.model.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.*;
import socialapp.userservice.model.enums.Gender;

import java.time.LocalDate;

public record RegistrationDto(
        @NotNull @Valid
        AddressDto address,
        @NotNull
        @NotBlank
        String firstName,
        @Email
        String email,
        @NotNull
        @NotBlank
        String lastName,
        @NotNull
        Gender gender,
        @PastOrPresent
        LocalDate birthDate
) {
}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/model/dto/RegistrationDto.kt
package socialapp.ktuserservice.model.dto

import jakarta.validation.Valid
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import socialapp.ktuserservice.model.entity.Address
import socialapp.ktuserservice.model.enums.Gender
import java.time.LocalDate

data class RegistrationDto(
    @field:NotNull
    @field:NotBlank
    val givenName: String,
    @field:NotNull
    @Valid
    val address: Address,
    @field:NotNull
    @field:NotBlank
    val familyName: String,
    @field:NotNull
    @field:NotBlank
    val preferredUsername: String,
    @field:Email
    val email: String,
    @field:NotNull
    val gender: Gender,
    val birthDate: LocalDate
)
